lib/Dialect/Torch/Transforms/DecomposeComplexOpsEarly.cpp

```diff
@@ -114,11 +110,6 @@ class DecomposeAtenSplitTensorLikeOp
       if (!matchPattern(op.getOperand(1), m_TorchConstantInt(&splitSize)))
         return rewriter.notifyMatchFailure(op, "split_size must be constant");
       chunks = (dimSize + splitSize - 1) / splitSize;
-    } else if (isa<AtenChunkOp>(op)) {
-      if (!matchPattern(op.getOperand(1), m_TorchConstantInt(&chunks)))
-        return rewriter.notifyMatchFailure(op,
-                                           "number of chunks must be constant");
-      splitSize = dimSize / chunks;
```

Comment:
This is not the right way of addressing my comment regarding templates and dynamic casts. There is still a dynamic cast in line https://github.com/llvm/torch-mlir/blob/705e4f698cc42d7a2e18eba6188785d8d91e1284/lib/Dialect/Torch/Transforms/DecomposeComplexOpsEarly.cpp#L109. There is also no longer valid support for the `AtenChunkOp`. You need to also add a decomposition of `AtenChunkOp` into `AtenSplitTensorOp`.